### PR TITLE
feat: Implement kat_messages (+ chore).

### PR DESCRIPTION
**Pull request by duesee**
This PR implements (an updated version of) the ["Messages" test vectors](https://github.com/mlswg/mls-implementations/blob/main/test-vectors.md#messages) from mls-implementations.

### CODE REVIEW

File: openmls/Cargo.toml [hunk 1]
```diff
@@ -46,15 +46,16 @@ crypto-debug = [] # ☣️ Enable logging of sensitive cryptographic information
 content-debug = [] # ☣️ Enable logging of sensitive message content
 
 [dev-dependencies]
+backtrace = "0.3"
 criterion = "^0.4"
-pretty_env_logger = "0.4"
+hex = { version = "0.4", features = ["serde"] }
```
Comment (duesee) [inline openmls/Cargo.toml:51]: This is a new dev-dependency. It allows to use ...

```rust
pub struct MessagesTestVector {
    #[serde(with = "hex::serde")]
    mls_welcome: Vec<u8>,
    // ...
```

... to get rid of hex-to-bytes conversions.

File: openmls/src/group/tests/kat_messages.rs [hunk 2]
```diff
@@ -684,28 +630,25 @@ pub fn run_test_vector(tv: MessagesTestVector) -> Result<(), MessagesTestVectorE
...
-//    let _tests: Vec<MessagesTestVector> = read("test_vectors/kat_messages.json");
-//
-//    for test_vector in tests {
-//        match run_test_vector(test_vector) {
-//            Ok(_) => {}
-//            Err(e) => panic!("Error while checking messages test vector.\n{:?}", e),
-//        }
-//    }
-//}
+#[test]
```
Comment (duesee) [inline openmls/src/group/tests/kat_messages.rs:637]: I've updated #1051accordingly.

File: openmls/src/group/tests/kat_messages.rs [hunk 3]
```diff
@@ -407,348 +398,297 @@ fn write_test_vectors_msg() {
...
-//            Ok(_) => {}
-//            Err(e) => panic!("Error while checking messages test vector.\n{:?}", e),
-//        }
-//    }
-//}
-
-/// Messages test vector error
+#[test]
+fn read_test_vectors_messages() {
+    let tests: Vec<MessagesTestVector> = read("test_vectors/kat_messages.json");
```
Comment (franziskuskiefer) [inline openmls/src/group/tests/kat_messages.rs:639]:
```suggestion
    let tests: Vec<MessagesTestVector> = read("test_vectors/messages.json");
```

File: openmls/src/group/tests/kat_messages.rs [hunk 4]
```diff
@@ -407,348 +398,297 @@ fn write_test_vectors_msg() {
     write("test_vectors/kat_messages-new.json", &tests);
```
Comment (franziskuskiefer) [inline openmls/src/group/tests/kat_messages.rs:398]:
```suggestion
    write("test_vectors/messages-new.json", &tests);
```

File: openmls/src/group/tests/kat_transcripts.rs [hunk 5]
```diff
@@ -26,18 +26,43 @@ use serde::{self, Deserialize, Serialize};
 use thiserror::Error;
 use tls_codec::{Deserialize as TlsDeserialize, Serialize as TlsSerializeTrait};
 
+/// ```json
```
Comment (franziskuskiefer) [inline openmls/src/group/tests/kat_transcripts.rs:29]: Unrelated? This is probably not worth touching because we should change it. Anyway, just a comment.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Only one nit really, let's use the same names as in the implementations repo. Otherwise it's really only changes in `kat_messages.rs` that we should just get in.

**Review by franziskuskiefer — APPROVED** [review]

**Pull request merged**